File: macos/SeedSeeker/Sources/SeedSeekerKit/DailyRunDate.swift
import Foundation

/// Date-picker presentation only; the Rust engine resolves daily seeds.
public enum DailyRunDate {
    public static var calendar: Calendar {
        var calendar = Calendar(identifier: .gregorian)
        calendar.timeZone = TimeZone(secondsFromGMT: 0)!
        return calendar
    }
    private static var formatter: DateFormatter {
        let formatter = DateFormatter()
        formatter.locale = Locale(identifier: "en_US_POSIX")
        formatter.calendar = calendar
        formatter.timeZone = calendar.timeZone
        formatter.dateFormat = "yyyy-MM-dd"
        formatter.isLenient = false
        return formatter
    }
    public static func code(_ date: Date = Date()) -> String { formatter.string(from: date) }
    public static func date(_ code: String) -> Date? { formatter.date(from: code) }
}

File: android/app/src/main/java/dev/seedseeker/app/model/DailyRunDate.kt
package dev.seedseeker.app.model

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

/** Calendar presentation only; the Rust engine resolves dates to daily seeds. */
object DailyRunDate {
    private fun formatter() = SimpleDateFormat("yyyy-MM-dd", Locale.ROOT).apply {
        timeZone = TimeZone.getTimeZone("UTC")
        isLenient = false
    }

    fun today(): String = format(System.currentTimeMillis())
    fun format(millis: Long): String = formatter().format(Date(millis))
    fun parse(date: String): Long? = runCatching {
        formatter().parse(date)?.time?.takeIf { format(it) == date }
    }.getOrNull()
}
